test/utils/introspection/SupportsInterface.behavior.js

```diff
@@ -111,18 +112,30 @@ function shouldSupportInterfaces(interfaces = []) {
       this.contractUnderTest = this.mock || this.token || this.holder || this.accessControl;
     });
 
-    it('supportsInterface uses less than 30k gas', async function () {
-      for (const k of interfaces) {
-        const interfaceId = INTERFACE_IDS[k] ?? k;
-        expect(await this.contractUnderTest.supportsInterface.estimateGas(interfaceId)).to.be.lte(30000);
-      }
+    describe('supportsInterface uses less than 30k gas', function () {
+      it('when it supports', async function () {
+        for (const k of interfaces) {
+          const interfaceId = INTERFACE_IDS[k] ?? k;
+          expect(await this.contractUnderTest.supportsInterface.estimateGas(interfaceId)).to.be.lte(30000);
+        }
+      });
+
+      it("when it doesn't support", async function () {
+        expect(await this.contractUnderTest.supportsInterface.estimateGas(InvalidID)).to.be.lte(30000);
+      });
     });
 
-    it('all interfaces are reported as supported', async function () {
-      for (const k of interfaces) {
-        const interfaceId = INTERFACE_IDS[k] ?? k;
-        expect(await this.contractUnderTest.supportsInterface(interfaceId)).to.equal(true, `does not support ${k}`);
-      }
+    describe('return values', async function () {
+      it('all interfaces are reported as supported', async function () {
+        for (const k of interfaces) {
+          const interfaceId = INTERFACE_IDS[k] ?? k;
+          expect(await this.contractUnderTest.supportsInterface(interfaceId)).to.equal(true, `does not support ${k}`);
+        }
+      });
+
+      it('invalidID is not supported', async function () {
+        expect(await this.contractUnderTest.supportsInterface(InvalidID)).to.be.equal(false, `supports ${InvalidID}`);
+      });
```

Comment:
I'd rather group these tests in a different way so that the "when ..." tests are `describe` or `context` blocks, for example:

```js
describe('when the interfaceId is supported', function () {
  it('uses less than 30k fas', function () { ... })

  it('returns true', function() { ... })
})

describe('when the interfaceId is not supported', function () {
  it('uses less thank 30k', function () { ... })

  it('returns false', function() { ... })
})
```

Reply:
Done